src/uk/co/llobile/kotlin/designpattern/behavioural/Observer.kts

```diff
@@ -0,0 +1,32 @@
+/**
+ * The observer pattern is a software design pattern in which an object, called the subject,
+ * maintains a list of its dependents, called observers, and notifies them automatically of
+ * any state changes, usually by calling one of their methods.
+ */
+
+// Example
+
+import kotlin.properties.Delegates
```

Comment:
The idea is to define the pattern using the classic names. In this case, they are `Subject` and `Observer`. Then I would create an example like the Color one.